src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Mapping/FacetMappingProfile.cs

```diff
@@ -1,58 +1,53 @@
 using System;
-using System.Collections.Generic;
 using System.Linq;
 using AutoMapper;
-using VirtoCommerce.Platform.Core.Common;
-using VirtoCommerce.SearchModule.Core.Model;
+using VirtoCommerce.CatalogModule.Core.Model.Search;
 using VirtoCommerce.XDigitalCatalog.Facets;
 
 namespace VirtoCommerce.XDigitalCatalog.Mapping
 {
-    public class MappingProfile : Profile
+    public class FacetMappingProfile : Profile
     {
-        public MappingProfile()
+        public FacetMappingProfile()
         {
-            CreateMap<AggregationRequest, FacetResult>().IncludeAllDerived();
+            CreateMap<Aggregation, FacetResult>().IncludeAllDerived();
 
-            CreateMap<TermAggregationRequest, TermFacetResult>().ConvertUsing((request, facet, context) =>
+            CreateMap<Aggregation, FacetResult>().ConvertUsing((request, facet, context) =>
             {
-                var aggregations = context.Items["aggregations"] as IList<AggregationResponse>;
-                var aggregation = aggregations.FirstOrDefault(x => x.Id.EqualsInvariant(request.Id) || x.Id.EqualsInvariant(request.FieldName));
-                if (aggregation != null)
+                FacetResult result = request.AggregationType switch
                 {
-                    return new TermFacetResult
+                    "attr" => new TermFacetResult
```

Comment:
I suppose you forgot for true `RangeFacetResult` types in response.
Try to test by using these arguments ` filter:"price:(TO 100)"` with this include `range_facet { name }`